gson/src/main/java/com/google/gson/internal/reflect/ReflectionHelper.java

```diff
@@ -111,34 +127,32 @@ private abstract static class RecordHelper {
     abstract String[] recordFields(Class<?> clazz);
 
     abstract <T> Constructor<T> getCanonicalRecordConstructor(Class<T> raw);
+
+    public abstract Method getAccessor(Class<?> raw, Field field);
   }
 
   private static class RecordSupportedHelper extends RecordHelper {
     private final Method isRecord;
     private final Method getRecordComponents;
     private final Method getName;
     private final Method getType;
+    private final Method getAccessor;
 
     private RecordSupportedHelper() throws NoSuchMethodException {
       isRecord = Class.class.getMethod("isRecord");
       getRecordComponents = Class.class.getDeclaredMethod("getRecordComponents");
       Class<?> recordComponentType = getRecordComponents.getReturnType().getComponentType();
       getName = recordComponentType.getDeclaredMethod("getName");
       getType = recordComponentType.getDeclaredMethod("getType");
+      getAccessor = recordComponentType.getDeclaredMethod("getAccessor");
     }
 
     @Override
     boolean isRecord(Class<?> raw) {
       try {
-        return boolean.class.cast(isRecord.invoke(raw));
-      } catch (IllegalAccessException e) {
-        throw createExceptionForUnexpectedIllegalAccess(e);
-      } catch (InvocationTargetException e) {
-        throw new RuntimeException("Failed to reflect into record components on"
-                + " class [" + raw + "], this is unexpected behaviour, as these methods should"
-                + " not throw when they are defined on Class",
-                e
-        );
+        return Boolean.class.cast(isRecord.invoke(raw)).booleanValue();
```

Comment:
I think this should work as well:
```suggestion
        return (boolean) isRecord.invoke(raw);
```

Reply:
@staale, this is not resolve yet, is it?